apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts

```diff
@@ -729,6 +792,43 @@ export class ReportsComponent implements OnInit {
         this.router.navigate(['/pages/inventory/v2', 'stock', this.moduleType?.toLowerCase(), 'edit', element?.stock?.uniqueName], { queryParams: { tab: 1 } });
     }
 
+    /**
+     * This will use for export stock transactions report data
+     *
+     * @return {*}  {void}
+     * @memberof ReportsComponent
+    */
+    public exportReport(): void {
+        setTimeout(() => {
+            this.isLoading = true;
+            if (this.reportType === InventoryReportType.stock) {
+
+                let stockReportRequestExport = this.stockReportRequestExport;
+                let queryParams = {
+                    from: this.fromDate,
+                    to: this.toDate,
+                };
+                delete stockReportRequestExport.from;
+                delete stockReportRequestExport.to;
+
+                stockReportRequestExport.inventoryType = this.moduleType;
+                this.inventoryService.getItemWiseReportExport(queryParams, stockReportRequestExport).pipe(takeUntil(this.cancelApi$)).subscribe(response => {
+                    this.isLoading = true;
```

Comment:
use component store if you are doing new task, do not use service in component basis

Reply:
done